src/models/message_data.rs

```diff
@@ -13,6 +13,9 @@ pub(crate) struct MessageData {
     /// Trace message
     pub(crate) message: LimitedLenString32768,
 
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) severity_level: Option<i32>,
```

Comment:
I'm confused that this is an `i32`. From https://github.com/Azure/azure-rest-api-specs/blob/e47988f3ccaa90681f11cce59c25c638ff305692/specification/applicationinsights/data-plane/Monitor.Exporters/preview/2020-09-15_Preview/swagger.json#L740-L750 I expected this to be a string. But `i32` probably leads to smaller HTTP requests, so that seems nice.

Do you think we could make an `enum SeverityLevel` for this to give the different values a name?

Reply:
I've made it an enum now, I tried to have it serializing as a None or as an Unit to be able to have the severity_level as a SeverityLevel type but when I use serialize_unit() on that property it ends up serialising a null instead. Maybe there's a better way to to that but I couldn't think of anything so I just created a method that we need to call populating the MessageData object.

Here's what my serialization was looking like just for reference:

```
//This will serialize it to an Application Insights compatible value
impl Serialize for SeverityLevel {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: Serializer,
    {
        match self {
            SeverityLevel::Verbose => serializer.serialize_i32(0),
            SeverityLevel::Debug | SeverityLevel::Information => serializer.serialize_i32(1),
            SeverityLevel::Warning => serializer.serialize_i32(2),
            SeverityLevel::Error => serializer.serialize_i32(3),
            SeverityLevel::None => serializer.serialize_unit(),
        }
    }
}

```